edb/pgsql/ast.py

```diff
@@ -394,17 +400,29 @@ class Query(ReturningQuery):
     """Generic superclass representing a query."""
 
     # Ignore the below fields in AST visitor/transformer.
-    __ast_meta__ = {'path_rvar_map',
+    __ast_meta__ = {'path_rvar_map', 'path_packed_rvar_map',
                     'view_path_id_map', 'argnames', 'nullable'}
 
     view_path_id_map: typing.Dict[irast.PathId, irast.PathId]
     # Map of RangeVars corresponding to paths.
     path_rvar_map: typing.Dict[typing.Tuple[irast.PathId, str], PathRangeVar]
+    # Map of materialized RangeVars corresponding to paths.
+    path_packed_rvar_map: typing.Dict[
+        typing.Tuple[irast.PathId, str],
+        PathRangeVar,
+    ]
 
     argnames: typing.Dict[str, Param]
 
     ctes: typing.List[CommonTableExpr]
 
+    def get_rvar_map(self, flavor: str) -> typing.Dict[
+            typing.Tuple[irast.PathId, str], PathRangeVar]:
+        if flavor == 'packed':
```

Comment:
Please use an explicit list of arms for `flavor` here with an `AssertionError` in else in case we add more of them in the future.

Reply:
Fun fact: for a while, I had a `semi-packed` flavor that behaved like `packed` in some places and `normal` in others. It was pretty bad.